pulsar-client/src/main/java/org/apache/pulsar/client/impl/ProducerImpl.java

```diff
@@ -492,6 +496,7 @@ public void sendAsync(Message<?> message, SendCallback callback) {
                 compressedPayload.release();
                 return;
             }
+            payloadChunkSize = Math.min(chunkMaxMessageSize, payloadChunkSize);
```

Comment:
I think we can remove the field `chunkMaxMessageSize` and use the value from conf directly.

Reply:
config can have default value `-1` and it needs to calculate right value based on user-configured value. so, we can't directly use from config.

Reply:
What I mean is check the config is enabled (!= -1) first.